src/controllers/post.js

```diff
@@ -1,39 +1,51 @@
 import dbClient from '../utils/dbClient.js'
-
 import { sendDataResponse } from '../utils/responses.js'
+import Joi from 'joi'
 
 export const create = async (req, res) => {
-  const { content } = req.body
-  if (!content) {
-    return sendDataResponse(res, 400, { content: 'Must provide content' })
+  const schema = Joi.object({
+    content: Joi.string().min(3).required()
+  })
+  const { error, value } = schema.validate(req.body)
+  if (error) {
+    return sendDataResponse(res, 400, { error: error.details[0].message })
   }
-  const createdPost = await dbClient.post.create({
-    data: {
-      content: content,
-      user: {
-        connect: {
-          id: req.user.id
+  const { content } = value
+  try {
+    const createdPost = await dbClient.post.create({
+      data: {
+        content: content,
+        user: {
+          connect: {
+            id: req.user.id
+          }
         }
       }
-    }
-  })
-  return sendDataResponse(res, 201, { post: createdPost })
+    })
+    return sendDataResponse(res, 201, { post: createdPost })
+  } catch (e) {
+    return sendDataResponse(res, 500, { error: e.message })
+  }
 }
 
 export const createComment = async (req, res) => {
-  const { content } = req.body
-  const { postId } = req.params
-
-  if (!content) {
-    return sendDataResponse(res, 400, { content: 'Must provide content' })
+  const schema = Joi.object({
```

Comment:
Nice! It's worth letting the other teams know about this - we'd want to to make sure we do validation in the same way across the server code base. You could offer to run a 10 minute demo for everyone showing how to use the library? It might also be worth a note in the README.

Reply:
great idea! I was thinking of this and I will do it InshaAllah